vertx-web/src/main/java/io/vertx/ext/web/impl/Origin.java

```diff
@@ -96,7 +97,7 @@ private Origin(String protocol, String host, String port, String resource) {
       if (!isValidChromeExtensionId(host, 0)) {
         throw new IllegalStateException("Illegal Chrome Extension id: " + host);
       }
-    } else {
+    } else if (!"moz-extension".equals(protocol)) {
```

Comment:
Can you please also apply a check for the extension format? Similar to what we have for Chrome extensions.